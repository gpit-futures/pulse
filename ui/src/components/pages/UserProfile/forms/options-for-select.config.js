const optionsForThemesField = [{
  value: 'green',
  title: 'Green Theme',
},
{
  value: 'purple',
  title: 'Purple Theme',
},
{
  value: 'gray',
  title: 'Gray Theme',
}];

const optionsForGenderField = [{
  value: 'female',
  title: 'Female',
}, {
  value: 'male',
  title: 'Male',
}];

const optionsForCountryField = [{
  value: 'usa',
  title: 'USA',
}, {
  value: 'country1',
  title: 'Country 1',
}, {
  value: 'country2',
  title: 'Country 2',
}, {
  value: 'country3',
  title: 'Country 3',
}, {
  value: 'country4',
  title: 'Country 4',
}, {
  value: 'country5',
  title: 'Country 5',
}];

export { optionsForThemesField, optionsForGenderField, optionsForCountryField }
