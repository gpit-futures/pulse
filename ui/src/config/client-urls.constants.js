import { themeConfigs } from '../themes.config';

export const clientUrls = {
  ROOT: '/',
  CHARTS: '/charts',
  PATIENTS: '/patients',
  PATIENTS_FULL_DETAILS: '/patients-full-details',
  SEARCH_REPORT: '/search-report',
  USER_PROFILE: '/profile',
  PATIENTS_SUMMARY: 'patients-summary',
  RESULTS: 'results',
  DIAGNOSES: 'diagnoses',
  ALLERGIES: 'allergies',
  CONTACTS: 'contacts',
  CLINICAL_NOTES: 'clinicalNotes',
  PERSONAL_NOTES: 'personalNotes',
  VACCINATIONS: 'vaccinations',
  MEDICATIONS: 'medications',
  GENERIC_PLUGIN: 'genericPlugin',
  PROCEDURES: 'procedures',
  EVENTS: 'events',
  TEST_RESULTS: 'results',
  REFERRALS: 'referrals',
  ORDERS: 'orders',
  MDTS: 'mdt',
  DRAWINGS: 'drawings',
  VITALS: 'vitals',
  DOCUMENTS: 'documents',
  IMAGES: 'images',
  TRANSFERS_OF_CARE: 'transfer-of-care',
  TOP_THREE_THINGS: 'topThreeThings',
  CLINICAL_STATEMENTS: 'clinicalStatements',
  PROMS: 'proms',
  FEEDS: 'feeds',
};

const pluginsPages = {
  'allergies': {
    breadcrumbs: [{
      title: 'Allergies',
      state: '/allergies',
    }],
  },
  'diagnoses': {
    breadcrumbs: [{
      title: 'Problems / Diagnoses',
      state: '/diagnoses',
    }],
  },
  'clinicalNotes': {
    breadcrumbs: [{
      title: 'Clinical Notes',
      state: '/clinicalNotes',
    }],
  },
  'personalNotes': {
    breadcrumbs: [{
      title: 'Personal Notes',
      state: '/personalNotes',
    }],
  },
  'contacts': {
    breadcrumbs: [{
      title: 'Contacts',
      state: '/contacts',
    }],
  },
  'vaccinations': {
    breadcrumbs: [{
      title: 'Vaccinations',
      state: '/vaccinations',
    }],
  },
  'genericPlugin': {
    breadcrumbs: [{
      title: 'Generic Plugin',
      state: '/genericPlugin',
    }],
  },
  'medications': {
    breadcrumbs: [{
      title: 'Medications',
      state: '/medications',
    }],
  },
  'procedures': {
    breadcrumbs: [{
      title: 'Procedures',
      state: '/procedures',
    }],
  },
  'events': {
    breadcrumbs: [{
      title: 'Events',
      state: '/events',
    }],
  },
  'results': {
    breadcrumbs: [{
      title: 'Test Results',
      state: '/results',
    }],
  },
  'referrals': {
    breadcrumbs: [{
      title: 'Referrals',
      state: '/referrals',
    }],
  },
  'orders': {
    breadcrumbs: [{
      title: 'Orders',
      state: '/orders',
    }],
  },
  'mdt': {
    breadcrumbs: [{
      title: 'MDT',
      state: '/mdt',
    }],
  },
  'drawings': {
    breadcrumbs: [{
      title: 'Drawings',
      state: '/drawings',
    }],
  },
  'vitals': {
    breadcrumbs: [{
      title: 'Vitals - NEWS',
      state: '/vitals',
    }],
  },
  'documents': {
    breadcrumbs: [{
      title: 'Documents',
      state: '/documents',
    }],
  },
  'images': {
    breadcrumbs: [{
      title: 'Images',
      state: '/images',
    }],
  },
  'transfer-of-care': {
    breadcrumbs: [{
      title: 'Transfers Of Care',
      state: '/transfers-of-care',
    }],
  },
  'topThreeThings': {
    breadcrumbs: [{
      title: 'Top 3 Things',
      state: '/topThreeThings',
    }],
  },
  'clinicalStatements': {
    breadcrumbs: [{
      title: 'Clinical Statements',
      state: '/clinicalStatements',
    }],
  },
  'proms': {
    breadcrumbs: [{
      title: 'PROMs',
      state: '/proms',
    }],
  },
};


const addPluginsPagesToLists = (list, breadcrumbsBefore) => {
  for (const nameOfPage in pluginsPages) {
    const breadcrumbs = breadcrumbsBefore.concat(pluginsPages[nameOfPage].breadcrumbs);
    list[nameOfPage] = { breadcrumbs };
  }
};

export const mainPagesTitles = {
  '/': {
    breadcrumbs: [{
      title: 'System Dashboard',
      state: '/',
    }],
    headerTitle: 'System Dashboard',
  },
  'charts': {
    breadcrumbs: [{
      title: 'System Dashboard',
      state: '/charts',
    }],
    headerTitle: 'System Dashboard',
  },
  'patients': {
    breadcrumbs: [{
      title: 'Home',
      state: '/',
    }, {
      title: 'Patient Listings',
      state: '/patients',
    }],
    headerTitle: 'Patients Lists',
  },
  'profile': {
    breadcrumbs: [{
      title: 'Patient Information',
      state: '/profile',
    }],
    headerTitle: 'Personal Information',
  },
  'search-report': {
    breadcrumbs: [{
      title: 'Home',
      state: '/',
    }, {
      title: 'Search Report',
      state: '/search-report',
    }],
    headerTitle: 'System Dashboard',
  },
  'patients-summary': {
    breadcrumbs: [{
      title: 'Patient Listings',
      state: '/patients',
    }, {
      title: themeConfigs.patientsSummaryPageName,
      state: '/patients-summary',
    }],
  },
  'patients-full-details': {
    breadcrumbs: [],
    headerTitle: 'Search results',
  },
};
addPluginsPagesToLists(mainPagesTitles, [{
  title: 'Patient Listings',
  state: '/patients',
}, {
  title: themeConfigs.patientsSummaryPageName,
  state: '/patients-summary',
}]);

export const mainPagesTitlesForPatients = {
  '/': {
    breadcrumbs: [{
      title: 'Patient Summary',
      state: '/patients-summary',
    }],
    headerTitle: 'Patient Summary',
  },

  'profile': {
    breadcrumbs: [{
      title: 'Patient Information',
      state: '/profile',
    }],
    headerTitle: 'Personal Information',
  },
  'patients-summary': {
    breadcrumbs: [{
      title: themeConfigs.patientsSummaryPageName,
      state: '/patients-summary',
    }],
  },
};
addPluginsPagesToLists(mainPagesTitlesForPatients, [{
  title: themeConfigs.patientsSummaryPageName,
  state: '/patients-summary',
}]);
