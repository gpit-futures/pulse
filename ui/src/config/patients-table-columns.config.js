import React from 'react';

import { getDDMMMYYYY } from '../utils/time-helpers.utils';

const CalendarIcon = () => <i className="fa fa-calendar" />;

export const patientsColumnsConfig = [
  { key: 'name', title: 'Name', width: 150 },
  { key: 'address', title: 'Address', width: 300 },
  { key: 'dateOfBirth', title: 'Born', transformer: getDDMMMYYYY, width: 105 },
  { key: 'promsCount', title: 'PROM ', icon: <span>#</span>, width: 100 },
  { key: 'promsDate', title: 'PROM', icon: <CalendarIcon />, transformer: getDDMMMYYYY, width: 110 },
  { key: 'gender', title: 'Gender', width: 90 },
  { key: 'id', title: 'NHS No.', width: 115 },
  { key: 'ordersDate', title: 'Orders', icon: <CalendarIcon />, transformer: getDDMMMYYYY, width: 110 },
  { key: 'ordersCount', title: 'Orders ', icon: <span>#</span>, width: 100 },
  { key: 'resultsDate', title: 'Results', icon: <CalendarIcon />, transformer: getDDMMMYYYY, width: 110 },
  { key: 'resultsCount', title: 'Results ', icon: <span>#</span>, width: 100 },
  { key: 'vitalsDate', title: 'Vitals', icon: <CalendarIcon />, transformer: getDDMMMYYYY, width: 110 },
  { key: 'vitalsCount', title: 'Vitals ', icon: <span>#</span>, width: 100 },
  { key: 'diagnosesDate', title: 'Diagnosis', icon: <CalendarIcon />, transformer: getDDMMMYYYY, width: 130 },
  { key: 'diagnosesCount', title: 'Diagnosis ', icon: <span>#</span>, width: 120 },
  { key: 'viewPatientNavigation', title: '', icon: <span />, width: 90 },
];

export const defaultColumnsSelected = {
  name: true,
  gender: true,
  dateOfBirth: true,
  address: true,
  id: true,
  ordersDate: true,
  resultsDate: true,
  vitalsDate: false,
  diagnosesDate: false,
  ordersCount: true,
  resultsCount: true,
  vitalsCount: false,
  diagnosesCount: false,
  viewPatientNavigation: true,
  promsCount: false,
  promsDate: false,
};
