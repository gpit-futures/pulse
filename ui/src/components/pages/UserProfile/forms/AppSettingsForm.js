import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import FileInput from '../../../form-fields/FileInput';
import { optionsForThemesField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validateAppSettingsForm } from '../forms.validation';
import { valuesSettingsForm, valuesSettingsFormLabels } from '../forms.config';
import { setTheme } from '../../../../ducks/set-theme.duck';

@reduxForm({
  form: 'appSettingsFormSelector',
  validate: validateAppSettingsForm,
})
export default class AppSettingsForm extends PureComponent {
  static propTypes = {
    dispatch: PropTypes.func,
    initialize: PropTypes.func,
  };

  componentDidMount() {
    const { patientsInfo, initialize } = this.props;
    initialize(this.defaultValuesForm(patientsInfo));
  }

  defaultValuesForm(value) {
    const defaultFormValues = {
      [valuesSettingsForm.APP_TITLE]: value.title,
      [valuesSettingsForm.LOGO_PATH]: value.logoB64,
      [valuesSettingsForm.SELECT_THEME]: value.themeColor,
      [valuesSettingsForm.BROWSER_TITLE]: value.browserTitle,
    };

    return defaultFormValues;
  }

 setThemeHook = dispatch => (theme) => {
   dispatch(setTheme(theme));
   return theme;
 };

 render() {
   const { dispatch } = this.props;

   return (
     <div className="panel-body-inner">
       <form name="appSettingsForm" className="form">
         <div className="form-group-wrapper">
           <Row>
             <Col xs={12} md={6}>
               <Row>
                 <Col md={11}>
                   <Field
                     label={valuesSettingsFormLabels.APP_TITLE}
                     name={valuesSettingsForm.APP_TITLE}
                     type="text"
                     placeholder=""
                     component={ValidatedInput}
                   />
                   <Field
                     label={valuesSettingsFormLabels.LOGO_PATH}
                     name={valuesSettingsForm.LOGO_PATH}
                     component={FileInput}
                     id={valuesSettingsForm.LOGO_PATH}
                   />
                   <Field
                     label={valuesSettingsFormLabels.SELECT_THEME}
                     name={valuesSettingsForm.SELECT_THEME}
                     component={Select}
                     options={optionsForThemesField}
                     //here we're using normalize hook for dispatching action to set app theme
                     normalize={this.setThemeHook(dispatch)}
                   />
                   <Field
                     label={valuesSettingsFormLabels.BROWSER_TITLE}
                     name={valuesSettingsForm.BROWSER_TITLE}
                     type="text"
                     placeholder=""
                     component={ValidatedInput}
                   />
                 </Col>
               </Row>
             </Col>
           </Row>
         </div>
       </form>
     </div>)
 }
}
