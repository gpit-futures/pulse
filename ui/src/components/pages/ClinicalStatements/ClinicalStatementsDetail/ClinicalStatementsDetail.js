import React, { PureComponent } from 'react';

import PluginDetailPanel from '../../../plugin-page-component/PluginDetailPanel';
import { Tag, TagList } from '../../../ui-elements/Tag/Tag';
import { getDDMMMYYYY } from '../../../../utils/time-helpers.utils';
import { valuesNames, valuesLabels } from '../forms.config';

const CLINICAL_STATEMENT_PANEL = 'clinicalStatementPanel';

export default class ClinicalStatementsDetail extends PureComponent {
  getTagList = (tags) => {
    const tagList = [];
    for (let key in tags) {
      tagList.push(<Tag key={`tag-${key}`} style="success">{key}</Tag>)
    }
    return tagList;
  };

  render() {
    const { onExpand, onShow, openedPanel, expandedPanel, currentPanel, onEdit, editedPanel, onCancel, onSaveSettings, clinicalStatementsDetailFormValues, metaPanelFormValues, isSubmit } = this.props;
    let { detail } = this.props;
    detail = detail || {};

    if (detail[valuesNames.STATEMENT]) {
      detail[valuesNames.STATEMENT] = detail[valuesNames.STATEMENT].replace(/fa-close|tag|editable|display: block;|display|block/gi, '');
    }
    const dateCreated = getDDMMMYYYY(detail[valuesNames.DATE_CREATED]);

    return (
      <div className="section-detail">
        <div className="panel-group accordion">
          {(expandedPanel === CLINICAL_STATEMENT_PANEL || expandedPanel === 'all') && !editedPanel[CLINICAL_STATEMENT_PANEL] ? <PluginDetailPanel
            onExpand={onExpand}
            name={CLINICAL_STATEMENT_PANEL}
            title="Clinical Statement"
            onShow={onShow}
            isOpen={openedPanel === CLINICAL_STATEMENT_PANEL}
            currentPanel={currentPanel}
            onEdit={onEdit}
            editedPanel={editedPanel}
            onCancel={onCancel}
            onSaveSettings={onSaveSettings}
            formValues={clinicalStatementsDetailFormValues}
            isBtnShowPanel={false}
            isShowControlPanel={false}
          >
            <div className="panel-body-inner">
              <div className="form">
                <div className="form-group-wrapper">
                  <div className="form-group">
                    <label className="control-label">{valuesLabels.TYPE}</label>
                    <div className="form-control-static">{detail[valuesNames.TYPE]}</div>
                  </div>

                  <div className="form-group">
                    <label className="control-label">{valuesLabels.STATEMENT}</label>
                    <div className="form-control-static" dangerouslySetInnerHTML={{ __html: detail[valuesNames.STATEMENT] }} />
                  </div>

                  <div className="form-group">
                    <label className="control-label">{valuesLabels.TAGS}</label>
                    <div className="form-control-static">
                      <TagList>
                        {this.getTagList(detail[valuesNames.TAGS])}
                      </TagList>
                     </div>
                  </div>

                  <div className="row-expand">
                    <div className="col-expand-left">
                      <div className="form-group">
                        <label className="control-label">{valuesLabels.DATE}</label>
                        <div className="form-control-static">{dateCreated}</div>
                      </div>
                    </div>
                    <div className="col-expand-right">
                      <div className="form-group">
                        <label className="control-label">{valuesLabels.AUTHOR}</label>
                        <div className="form-control-static">{detail[valuesNames.AUTHOR]}</div>
                      </div>
                    </div>
                  </div>


                  <div className="form-group">
                    <label className="control-label">{valuesLabels.SOURCE}</label>
                    <div className="form-control-static">{detail[valuesNames.SOURCE]}</div>
                  </div>
                </div>
              </div>
            </div>
          </PluginDetailPanel> : null}
        </div>
      </div>
    )
  }
}
