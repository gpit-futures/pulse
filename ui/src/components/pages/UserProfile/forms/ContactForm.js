import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import { optionsForCountryField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validateContactForm } from '../forms.validation';
import { valuesContactForm, valuesContactFormLabels } from '../forms.config';
import { defaultContactFormValues } from './default-values.config';

@reduxForm({
  form: 'contactFormSelector',
  validate: validateContactForm,
})
export default class ContactForm extends PureComponent {
  componentDidMount() {
    this.props.initialize(defaultContactFormValues);
  }

  render() {
    return (
      <div className="panel-body-inner">
        <form name="contactForm" className="form">
          <div className="form-group-wrapper">
            <Row>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11}>
                    <Field
                      label={valuesContactFormLabels.ADDRESS}
                      name={valuesContactForm.ADDRESS}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.CITY}
                      name={valuesContactForm.CITY}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.STATE}
                      name={valuesContactForm.STATE}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.POSTAL_CODE}
                      name={valuesContactForm.POSTAL_CODE}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.SELECT_COUNTRY}
                      name={valuesContactForm.SELECT_COUNTRY}
                      placeholder=""
                      component={Select}
                      options={optionsForCountryField}
                    />
                  </Col>
                </Row>
              </Col>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11} mdOffset={1}>
                    <Field
                      label={valuesContactFormLabels.PHONE}
                      name={valuesContactForm.PHONE}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.EMAIL}
                      name={valuesContactForm.EMAIL}
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
            </Row>
          </div>
        </form>
      </div>)
  }
}
