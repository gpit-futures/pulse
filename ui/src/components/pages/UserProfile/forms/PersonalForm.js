import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import DateInput from '../../../form-fields/DateInput';
import { optionsForGenderField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validatePersonalForm } from '../forms.validation';
import { valuesPersonalForm, valuesPersonalFormLabels } from '../forms.config';
import { defaultPersonalFormValues } from './default-values.config';

@reduxForm({
  form: 'personalFormSelector',
  validate: validatePersonalForm,
})
export default class PersonalForm extends PureComponent {
  componentDidMount() {
    this.props.initialize(defaultPersonalFormValues);
  }

  render() {
    return (
      <div className="panel-body-inner">
        <form name="personalForm" className="form">
          <div className="form-group-wrapper">
            <Row>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11}>
                    <Field
                      label={valuesPersonalFormLabels.FIRST_NAME}
                      name={valuesPersonalForm.FIRST_NAME}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesPersonalFormLabels.LAST_NAME}
                      name={valuesPersonalForm.LAST_NAME}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesPersonalFormLabels.NHS_NUMBER}
                      name={valuesPersonalForm.NHS_NUMBER}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11} mdOffset={1}>
                    <Field
                      label={valuesPersonalFormLabels.DATE_OF_BIRTH}
                      name={valuesPersonalForm.DATE_OF_BIRTH}
                      type="text"
                      placeholder=""
                      component={DateInput}
                      props={{ format: 'DD-MMM-YYYY' }}
                    />
                    <Field
                      label={valuesPersonalFormLabels.SELECT_GENDER}
                      name={valuesPersonalForm.SELECT_GENDER}
                      placeholder=""
                      component={Select}
                      options={optionsForGenderField}
                    />
                    <Field
                      label={valuesPersonalFormLabels.DOCTOR}
                      name={valuesPersonalForm.DOCTOR}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
            </Row>
          </div>
        </form>
      </div>)
  }
}
