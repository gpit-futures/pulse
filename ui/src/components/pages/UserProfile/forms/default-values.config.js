import { valuesSettingsForm, valuesPersonalForm, valuesContactForm } from '../forms.config';

const defaultPersonalFormValues = {
  [valuesPersonalForm.FIRST_NAME]: 'Bob',
  [valuesPersonalForm.LAST_NAME]: 'Smith',
  [valuesPersonalForm.SELECT_GENDER]: 'female',
  [valuesPersonalForm.DOCTOR]: 'Dr Emma Huston',
  [valuesPersonalForm.DATE_OF_BIRTH]: 0,
};

const defaultContactFormValues = {
  [valuesContactForm.ADDRESS]: '6801 Tellus Street',
  [valuesContactForm.CITY]: 'Westmorland',
  [valuesContactForm.STATE]: 'Westmorland',
  [valuesContactForm.POSTAL_CODE]: 'Box 306',
  [valuesContactForm.PHONE]: '07624 647524',
  [valuesContactForm.EMAIL]: 'bob.smith@gmail.com',
};

const defaultAppSettingsFormValues = {
  [valuesSettingsForm.APP_TITLE]: 'should be fetched from server',
  [valuesSettingsForm.LOGO_PATH]: 'should be fetched from server',
  [valuesSettingsForm.SELECT_THEME]: 'themeDefault',
  [valuesSettingsForm.BROWSER_TITLE]: 'should be fetched from server',
};

export { defaultPersonalFormValues, defaultContactFormValues, defaultAppSettingsFormValues }
