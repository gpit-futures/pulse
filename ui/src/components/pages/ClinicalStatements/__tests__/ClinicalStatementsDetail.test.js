import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';

import ClinicalStatementsDetail from '../ClinicalStatementsDetail/ClinicalStatementsDetail';
import { valuesNames, valuesLabels } from '../forms.config';
import { getDDMMMYYYY } from '../../../../utils/time-helpers.utils';

Enzyme.configure({ adapter: new Adapter() });

const testProps = {
  detail: {
    [valuesNames.TAGS]: { chestpain: true, shouder: true },
    [valuesNames.STATEMENT]: '<span class="tag" data-tag-id="1519236197404" data-id="7" data-phrase="The pain was mild at |?| in severity" contenteditable="false">The pain was mild at <span class="editable editable-click editable-unsaved editable-empty editable-open" contenteditable="false" data-arr-subject="The pain was mild at " editable-text="" data-arr-unit="?" data-arr-value=" in severity" data-original-title="" title="" style="background-color: rgba(0, 0, 0, 0);" aria-describedby="popover368681">Empty</span><div class="popover editable-container editable-popup fade top in" role="tooltip" id="popover368681" style="top: -74px; left: 12.6719px; display: block;"><div class="arrow" style="left: 50%;"></div><h3 class="popover-title">Edit Text</h3><div class="popover-content"> <div><div class="editableform-loading" style="display: none;"></div><form class="form-inline editableform" style=""><div class="control-group form-group"><div><div class="editable-input" style="position: relative;"><input type="text" class="form-control input-sm" style="padding-right: 24px;"><span class="editable-clear-x" style=""></span></div><div class="editable-buttons"><button type="submit" class="btn btn-primary btn-sm editable-submit"><i class="glyphicon glyphicon-ok"></i></button><button type="button" class="btn btn-default btn-sm editable-cancel"><i class="glyphicon glyphicon-remove"></i></button></div></div><div class="editable-error-block help-block" style="display: none;"></div></div></form></div></div></div> in severity. <a class="remove" contenteditable="false"><i class="fa fa-close" contenteditable="false"></i></a></span> <span id="temp" contenteditable="false"></span>',
    [valuesNames.TYPE]: 'Test name',
    [valuesNames.AUTHOR]: 'bob.smith@gmail.com',
    [valuesNames.DATE_CREATED]: 1519236225009,
    [valuesNames.SOURCE]: 'qewdDB',
    [valuesNames.SOURCE_ID]: 'd9acb27f-6688-4515-ab0a-5db636f36f05',
  },
};

const CLINICAL_STATEMENT_PANEL = 'clinicalStatementPanel';
const CONVERT_DATE = getDDMMMYYYY(testProps.detail[valuesNames.DATE_CREATED]);

describe('Component <ClinicalStatementsDetail />', () => {
  it('should renders with props correctly', () => {
    const component = shallow(<ClinicalStatementsDetail />);

    // Testing component when detail filled object, expandedPanel is all, and panel not edited
    component.setProps({ detail: testProps.detail, expandedPanel: 'all', editedPanel: { [CLINICAL_STATEMENT_PANEL]: false } });
    expect(component.props().className).toEqual('section-detail');
    expect(component.find('PluginDetailPanel')).toHaveLength(1);

    // Testing clinicalStatementsPanel
    expect(component.find('PluginDetailPanel').at(0).props().name).toEqual(CLINICAL_STATEMENT_PANEL);
    expect(component.find('PluginDetailPanel').at(0).props().title).toEqual('Clinical Statement');
    expect(component.find('PluginDetailPanel').at(0).props().isOpen).toEqual(false);
    expect(component.find('PluginDetailPanel').at(0).props().isBtnShowPanel).toEqual(false);
    expect(component.find('PluginDetailPanel').at(0).props().isShowControlPanel).toEqual(false);

    expect(component.find('.control-label').at(0).text()).toEqual(valuesLabels.TYPE);
    expect(component.find('.control-label').at(1).text()).toEqual(valuesLabels.STATEMENT);
    expect(component.find('.control-label').at(2).text()).toEqual(valuesLabels.TAGS);
    expect(component.find('.control-label').at(3).text()).toEqual(valuesLabels.DATE);
    expect(component.find('.control-label').at(4).text()).toEqual(valuesLabels.AUTHOR);

    const formGroups = component.find('.form-group');
    expect(formGroups).toHaveLength(6);
    expect(formGroups.at(0).find('.form-control-static').text()).toEqual(testProps.detail[valuesNames.TYPE]);
    expect(formGroups.at(1).find('.form-control-static').html()).toEqual(`<div class="form-control-static">${testProps.detail[valuesNames.STATEMENT]}</div>`);
    const formControlStatic3 = formGroups.at(2).find('.form-control-static');
    expect(formControlStatic3.find('TagList')).toHaveLength(1);
    expect(formControlStatic3.find('Tag')).toHaveLength(2);
    expect(formGroups.at(3).find('.form-control-static').text()).toEqual(CONVERT_DATE);
    expect(formGroups.at(4).find('.form-control-static').text()).toEqual(testProps.detail[valuesNames.AUTHOR]);
    expect(formGroups.at(5).find('.form-control-static').text()).toEqual(testProps.detail[valuesNames.SOURCE]);

    expect(component).toMatchSnapshot();
  });
});

