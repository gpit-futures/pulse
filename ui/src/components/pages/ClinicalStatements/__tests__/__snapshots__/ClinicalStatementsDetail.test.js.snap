// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <ClinicalStatementsDetail /> should renders with props correctly 1`] = `
<div
  className="section-detail"
>
  <div
    className="panel-group accordion"
  >
    <PluginDetailPanel
      editedPanel={
        Object {
          "clinicalStatementPanel": false,
        }
      }
      isBtnShowPanel={false}
      isOpen={false}
      isShowControlPanel={false}
      name="clinicalStatementPanel"
      title="Clinical Statement"
    >
      <div
        className="panel-body-inner"
      >
        <div
          className="form"
        >
          <div
            className="form-group-wrapper"
          >
            <div
              className="form-group"
            >
              <label
                className="control-label"
              >
                Type
              </label>
              <div
                className="form-control-static"
              >
                Test name
              </div>
            </div>
            <div
              className="form-group"
            >
              <label
                className="control-label"
              >
                Statement
              </label>
              <div
                className="form-control-static"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"\\" data--id=\\"1519236197404\\" data-id=\\"7\\" data-phrase=\\"The pain was mild at |?| in severity\\" content=\\"false\\">The pain was mild at <span class=\\" -click -unsaved -empty -open\\" content=\\"false\\" data-arr-subject=\\"The pain was mild at \\" -text=\\"\\" data-arr-unit=\\"?\\" data-arr-value=\\" in severity\\" data-original-title=\\"\\" title=\\"\\" style=\\"background-color: rgba(0, 0, 0, 0);\\" aria-describedby=\\"popover368681\\">Empty</span><div class=\\"popover -container -popup fade top in\\" role=\\"tooltip\\" id=\\"popover368681\\" style=\\"top: -74px; left: 12.6719px; \\"><div class=\\"arrow\\" style=\\"left: 50%;\\"></div><h3 class=\\"popover-title\\">Edit Text</h3><div class=\\"popover-content\\"> <div><div class=\\"form-loading\\" style=\\": none;\\"></div><form class=\\"form-inline form\\" style=\\"\\"><div class=\\"control-group form-group\\"><div><div class=\\"-input\\" style=\\"position: relative;\\"><input type=\\"text\\" class=\\"form-control input-sm\\" style=\\"padding-right: 24px;\\"><span class=\\"-clear-x\\" style=\\"\\"></span></div><div class=\\"-buttons\\"><button type=\\"submit\\" class=\\"btn btn-primary btn-sm -submit\\"><i class=\\"glyphicon glyphicon-ok\\"></i></button><button type=\\"button\\" class=\\"btn btn-default btn-sm -cancel\\"><i class=\\"glyphicon glyphicon-remove\\"></i></button></div></div><div class=\\"-error- help-\\" style=\\": none;\\"></div></div></form></div></div></div> in severity. <a class=\\"remove\\" content=\\"false\\"><i class=\\"fa \\" content=\\"false\\"></i></a></span> <span id=\\"temp\\" content=\\"false\\"></span>",
                  }
                }
              />
            </div>
            <div
              className="form-group"
            >
              <label
                className="control-label"
              >
                Tags
              </label>
              <div
                className="form-control-static"
              >
                <TagList>
                  <Tag
                    key="tag-chestpain"
                    style="success"
                  >
                    chestpain
                  </Tag>
                  <Tag
                    key="tag-shouder"
                    style="success"
                  >
                    shouder
                  </Tag>
                </TagList>
              </div>
            </div>
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Date
                  </label>
                  <div
                    className="form-control-static"
                  >
                    21-Feb-2018
                  </div>
                </div>
              </div>
              <div
                className="col-expand-right"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Author
                  </label>
                  <div
                    className="form-control-static"
                  >
                    bob.smith@gmail.com
                  </div>
                </div>
              </div>
            </div>
            <div
              className="form-group"
            >
              <label
                className="control-label"
              >
                Source
              </label>
              <div
                className="form-control-static"
              >
                qewdDB
              </div>
            </div>
          </div>
        </div>
      </div>
    </PluginDetailPanel>
  </div>
</div>
`;
